src/Microsoft.ML.Tokenizers/Microsoft.ML.Tokenizers.csproj

```diff
@@ -21,4 +21,38 @@
     <PackageReference Include="System.Text.Json" Version="$(SystemTextJsonVersion)" />
   </ItemGroup>
 
+  <UsingTask TaskName="CompressFile"
+    TaskFactory="RoslynCodeTaskFactory"
+    AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll" >
+    <ParameterGroup>
+      <Files ParameterType="Microsoft.Build.Framework.ITaskItem[]" Required="true" />
+    </ParameterGroup>
+    <Task>
+      <Using Namespace="System.IO" />
+      <Using Namespace="System.IO.Compression" />
+      <Code Type="Fragment" Language="cs">
+        foreach(var file in Files)
+        {
+            using var sourceStream = File.OpenRead(file.GetMetadata("FullPath"));
+            using var destStream = new DeflateStream(File.Create(file.GetMetadata("Destination")), CompressionLevel.Optimal);
+            sourceStream.CopyTo(destStream);
+        }
+      </Code>
+    </Task>
+  </UsingTask>
+
+  <ItemGroup>
```

Comment:
Do we need to give attribution for these files, mention where we downloaded them from and how to update?

Reply:
I added the Source attribute to every added file element.

Reply:
Sorry, by attribution I didn't literally mean add an attribute.  I meant "OSS Attribution" or "give credit or show the source of a particular file".  More like a comment:
```
<!-- These files were downloaded from XXXX.
        Copyright licenses can be found in YYYY.
       After being downloaded they were modified in this way ...  -->
```